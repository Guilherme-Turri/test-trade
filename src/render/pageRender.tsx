import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};