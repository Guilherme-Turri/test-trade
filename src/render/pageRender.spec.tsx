import { screen } from '@testing-library/react';
import { pageRender } from './pageRender';

describe('pageRender', () => {
  it('renders children componenet', () => {
    const ChildComponent = () => <p data-testid="child">Child</p>;
    const { container } = pageRender(<ChildComponent />);
    const child = screen.getByTestId('child');
    expect(child).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
