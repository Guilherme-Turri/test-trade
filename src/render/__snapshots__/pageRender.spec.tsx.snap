// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pageRender renders children componenet 1`] = `
<div>
  <p
    data-testid="child"
  >
    Child
  </p>
</div>
`;
